### ISSUE TITLE
Improve S3900: Group raised issues referring to the same parameter

### ISSUE THREAD

**Issue by zsolt-kolbay-sonarsource** [SonarSource/sonar-dotnet#7135]
Currently, S3900 can raise an issue multiple times for the same method argument if it's being dereferenced in different branches:

```cs
public void Method(object o)
{
   if (...) 
   {
      var s = o.ToString(); // Noncompliant
   }
   else if (...)
   {
      var s = o.ToString(); // Noncompliant
   }
   else
   {
      var s = o.ToString(); // Noncompliant
   }
}
```

The rule is quite noisy - [over 80000 issues in Peach](https://peach.sonarsource.com/coding_rules?languages=cs&q=S3900&open=csharpsquid%3AS3900) - and null check is usually done only once per every parameter, at the beginning of the method.

Raise only one issue per method argument, and each place where it's dereferenced should be added as a location to that single issue.
So the previous example should look like this:

```cs
public void Method(object o) // Noncompliant - issue is only raised here
{
   if (...) 
   {
      var s = o.ToString(); // issue location 1
   }
   else if (...)
   {
      var s = o.ToString(); // issue location 2
   }
   else
   {
      var s = o.ToString(); // issue location 3
   }
}
```

**Comment by pavel-mikula-sonarsource** [comment]
The problem with this is, that Roslyn and SonarLint do not show these issues. So as soon as user fixes the first one inside the 'if', he'll get a new issue on the next line. And that's frustrating experience.
```
public void Method(object o) // Noncompliant - issue is only raised here
{
   if (...) 
   {
     if(o is null) 
        return;
      var s = o.ToString(); 
   }
   else if (...)
   {
      var s = o.ToString(); // new issue
   }
   else
   {
      var s = o.ToString(); // secondary location
   }
}
```

At the same time, we wouldn't be able to raise issues that we are able to find, but we are not able to analyze the whole method because it's too big.

As this was also the behavior of the old rule too and nobody complained about it, I wouldn't change the behavior.
